packages/data-stores/src/domain-suggestions/selectors.ts

```diff
@@ -7,127 +7,80 @@ import { select } from '@wordpress/data';
-		_state: State,
-		search: string,
-		options: DomainSuggestionSelectorOptions = {}
-	) {
-		const normalizedQuery = normalizeDomainSuggestionQuery( search, options );
-
-		return select( 'core/data' ).isResolving( STORE_KEY, '__internalGetDomainSuggestions', [
-			normalizedQuery,
-		] );
-	}
-
-	/**
-	 * Normalize domain query
-	 *
-	 * It's important to have a consistent, reproduceable representation of a domains query so that the result can be
-	 * stored and retrieved.
-	 *
-	 * @see client/state/domains/suggestions/utils.js
-	 * @see client/components/data/query-domains-suggestions/index.jsx
-	 *
-	 * @param search       Domain search string
-	 * @param queryOptions Optional paramaters for the query
-	 * @returns Normalized query object
-	 */
-	function normalizeDomainSuggestionQuery(
-		search: string,
-		queryOptions: DomainSuggestionSelectorOptions
-	): DomainSuggestionQuery {
-		return {
-			// Defaults
-			include_wordpressdotcom: queryOptions.only_wordpressdotcom || false,
-			include_dotblogsubdomain: false,
-			only_wordpressdotcom: false,
-			quantity: 5,
-			vendor,
+import { stringifyDomainQueryObject, normalizeDomainSuggestionQuery } from './utils';
+
+export const getCategories = ( state: State ): DomainCategory[] => {
+	// Sort domain categories by tier, then by title.
+	return [
+		...state.categories
+			.filter( ( { tier } ) => tier !== null )
+			.sort( ( a, b ) => ( a > b ? 1 : -1 ) ),
+		...state.categories
+			.filter( ( { tier } ) => tier === null )
+			.sort( ( a, b ) => a.title.localeCompare( b.title ) ),
+	];
+};
 
-			// Merge options
-			...queryOptions,
+export const getDomainSuggestions = (
+	_state: State,
+	search: string,
+	options: DomainSuggestionSelectorOptions = {}
+): DomainSuggestion[] | undefined => {
+	const normalizedQuery = normalizeDomainSuggestionQuery( search, options );
 
-			// Add the search query
-			query: search.trim().toLocaleLowerCase(),
-		};
-	}
+	// We need to go through the `select` store to get the resolver action
+	return select( STORE_KEY ).__internalGetDomainSuggestions( normalizedQuery );
+};
 
-	/**
-	 * Do not use this selector. It is for internal use.
-	 *
-	 * @param state Store state
-	 * @param queryObject Normalized object representing the query
-	 * @returns suggestions
-	 */
-	function __internalGetDomainSuggestions( state: State, queryObject: DomainSuggestionQuery ) {
-		return state.domainSuggestions.data[ stringifyDomainQueryObject( queryObject ) ];
-	}
+export const getDomainState = ( state: State ): DataStatus => {
+	return state.domainSuggestions.state;
+};
 
-	function isAvailable( state: State, domainName: string ) {
-		return state.availability[ domainName ];
-	}
+export const getDomainErrorMessage = ( state: State ): string | null => {
+	return state.domainSuggestions.errorMessage;
+};
 
-	function getDomainAvailabilities( state: State ) {
-		return state.availability;
-	}
+// TODO: reconcile this function with "Pending" status?
+// It doesn't seem to be used
```

Comment:
These 2 lines of comments are not from me. They were here before. Won't be covering in this PR.